src/sentinel.c

```diff
@@ -4987,7 +4983,7 @@ sentinelRedisInstance *sentinelSelectSlave(sentinelRedisInstance *master) {
     di = dictGetIterator(master->slaves);
 
     while((de = dictNext(di)) != NULL) {
-        sentinelRedisInstance *slave = dictGetVal(de);
+        slave = dictGetVal(de);
```

Comment:
I think it is better to limit the declration of `slave` to this block and not declare it at the begining of the function.

Reply:
Updated to limit `slave` inside the `while loop`, as what the origin code do